libbeat/outputs/redis/redis.go

```diff
@@ -129,8 +132,23 @@ func makeRedis(
 			return outputs.Fail(err)
 		}
 
+		pass := config.Password
+		if parts := strings.SplitN(host, "://", 2); len(parts) != 2 {
+			// Add scheme.
+			host = fmt.Sprintf("redis://%s", host)
+		}
+		url, err := url.Parse(host)
+		if err != nil {
+			return outputs.Fail(err)
+		}
+
+		hostPass, passSet := url.User.Password()
+		if passSet {
+			pass = hostPass
+		}
```

Comment:
This would force the user to always prepend with `:`. If we want to get rid of the `:` prefix then it becomes as easy as:

```

```

and with optional `:` prefix:

```
var pass string
if host[0] == ':' {
  host = host[1:]
}
if parts := strings.SplitN(host, "@" 2); len(parts) == 2 {
  pass = parts[0]
  host = parts[1]
}
```

Reply:
Sure I understand but my concern is `pass@host:1234` is not a valid url because `pass` here would be the username for any url parser, so it could be confusing for any user I think. 

But if you  are ok with this, I'll apply your suggestions @urso :+1: